src/lib/utils/options/readers/tsconfig.ts

```diff
@@ -82,7 +82,7 @@ export class TSConfigReader extends OptionsComponent
             delete compilerOptions[key];
         }
 
-        _.merge(event.data, compilerOptions);
+        _.defaultsDeep(event.data, compilerOptions);
```

Comment:
I chose `defaultsDeep` because I thought it might be more future proof. Let me know if you would like the PR amended.